plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/tools/PostgreBackupAllWizardPageObjects.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+    @Override
+    protected boolean determinePageCompletion() {
+        boolean complete = false;
+        if (!checkedObjects.isEmpty()) {
+            complete = true;
+        }
+        for (TableItem item : databasesTable.getItems()) {
+            if (item.getChecked()) {
+                complete = true;
+                break;
+            }
+        }
+        return complete && super.determinePageCompletion();
+    }
+
+    @Override
+    public void activatePage() {
+        super.activatePage();
+        loadSettings();
+        updateState();
+    }
+
+    @Override
+    public void deactivatePage() {
+        saveState();
+    }
+
+    private void loadSettings() {
+        checkedObjects.clear();
+        databasesTable.removeAll();
+
+        Set<PostgreDatabase> activeDatabases = new LinkedHashSet<>();
+        for (PostgreDatabaseBackupAllInfo info : wizard.getSettings().getExportObjects()) {
+            dataSource = info.getDataSource();
+            List<PostgreDatabase> databases = info.getDatabases();
+            if (!CommonUtils.isEmpty(databases)) {
+                activeDatabases.addAll(databases);
+            }
+        }
+        if (dataSource != null) {
+            // Database list depends on connection setting
+            for (PostgreDatabase database : dataSource.getDatabases()) {
+                TableItem item = new TableItem(databasesTable, SWT.NONE);
+                item.setImage(DBeaverIcons.getImage(DBIcon.TREE_DATABASE));
+                item.setText(0, database.getName());
+                item.setData(database);
+                if (activeDatabases.contains(database)) {
+                    item.setChecked(true);
+                    databasesTable.select(databasesTable.indexOf(item));
+                }
+            }
+        }
+    }
+
+    @Override
+    public void saveState() {
+        super.saveState();
+        List<PostgreDatabaseBackupAllInfo> objects = wizard.getSettings().getExportObjects();
+        objects.clear();
+        List<PostgreDatabase> databases = new ArrayList<>();
+        for (TableItem item : databasesTable.getItems()) {
+            if (item.getChecked()) {
+                PostgreDatabase database = (PostgreDatabase) item.getData();
+                if (!databases.contains(database)) {
+                    databases.add(database);
+                }
+            }
+        }
+        PostgreDatabaseBackupAllInfo info = new PostgreDatabaseBackupAllInfo(dataSource, databases);
+        objects.add(info);
+    }
+
+    @Override
+    protected void updateState() {
+        updatePageCompletion();
+        getContainer().updateButtons();
+    }
+
+    @Override
+    protected void updateTableCheckedStatus(@NotNull Table table, boolean check) {
+        // Handle event from buttons "All" and "None"
+        if (table == databasesTable) {
+            TableItem[] items = databasesTable.getItems();
+            if (items.length != 0) {
```

Comment:
Redundant check; the variable can also be inlined if check is removed

Reply:
Why not? The compiler expands array iteration like so:

```java
TableItem[] items = databasesTable.getItems();
for (int i = 0; i < items.length; i++) {
    ...
}
```

So it checks whether the number of elements in that array is greater than zero. You're doing the job you shouldn't be doing.